examples/validation/src/getting_started_validator_decorator_function.py

```diff
@@ -0,0 +1,28 @@
+from dataclasses import dataclass, field
+from uuid import uuid4
+
+import getting_started_validator_decorator_schema as schemas
+
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from aws_lambda_powertools.utilities.validation import validator
+
+
+@dataclass
+class User:
+    ip: str
+    permissions: list
+    user_id: str = field(default_factory=lambda: f"{uuid4()}")
+    name: str = "Project Lambda Powertools"
+
+
+# using a decorator to validate input and output data
+@validator(inbound_schema=schemas.INPUT, outbound_schema=schemas.OUTPUT)
+def lambda_handler(event, context: LambdaContext) -> dict:
+
+    user_details: dict = {}
+
+    if event.get("username") == "lambda" and event.get("password") == "powertools":
+        user_details = User(ip=event.get("ip"), permissions=["read", "write"]).__dict__
+
+    # the body must be a object because must match OUTPUT schema, otherwise it fails
+    return {"body": user_details or None, "statusCode": 200 if user_details else 204}
```

Comment:
Consider revisiting username/password idea, maybe get users instead? It's a highly visited doc and we could accidentally give the wrong impression that this is a good security mechanism.

Reply:
Nice! I'll refactor this here and in the other one example.